zone/tradeskills.cpp

```diff
@@ -1866,37 +1866,53 @@ bool ZoneDatabase::DisableRecipe(uint32 recipe_id)
 bool Client::CheckTradeskillLoreConflict(int32 recipe_id)
 {
 	auto recipe_entries = TradeskillRecipeEntriesRepository::GetWhere(
-			content_db,
-			fmt::format(
-					"recipe_id = {} ORDER BY componentcount DESC",
-					recipe_id
-			)
+		content_db,
+		fmt::format(
+			"recipe_id = {} ORDER BY componentcount DESC",
+			recipe_id
+		)
 	);
 	if (recipe_entries.empty()) {
 		return false;
 	}
 
-	for (auto& f : recipe_entries) {
-		for (auto &e: recipe_entries) {
-			if (f.componentcount > 0 && e.item_id == f.item_id && e.componentcount == 0) {
-				e.item_id = 0;
+	const EQ::ItemData* fi = nullptr;
+	const EQ::ItemData* ei = nullptr;
+	for (auto &f : recipe_entries) {
+		if (f.item_id) {
+			fi = database.GetItem(f.item_id);
+			for (auto &e : recipe_entries) {
+				bool fi_is_valid = e.item_id && fi && fi->LoreGroup != 0;
+				bool item_id_match = e.item_id == f.item_id;
+
+				if (fi_is_valid) {
+					ei = database.GetItem(e.item_id);
+					bool ei_is_valid = ei && ei->LoreGroup != 0;
+					bool unique_lore_group_match = fi->LoreGroup > 0 && fi->LoreGroup == ei->LoreGroup;
```

Comment:
This is a great variable name, tells me exactly what's happening

Reply:
thanks!